sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -157,10 +158,17 @@ public DigitalTwinsServiceVersion getServiceVersion() {
         return protocolLayer
             .getDigitalTwins()
             .addWithResponseAsync(digitalTwinId, digitalTwin, context)
-            .map(response -> {
-                T genericResponse = mapper.convertValue(response.getValue(), clazz);
+            .flatMap(response -> {
+                T genericResponse = null;
+                try {
+                    genericResponse = DeserializationHelpers.castObject(mapper, response.getValue(), clazz);
+                } catch (JsonProcessingException e) {
+                    logger.error("JsonProcessingException occurred while creating digital twin: ", e);
```

Comment:
nit: maybe say that the exception occurred while deserializing the response -> since technically the twin might have been created successfully, but the CL deserialization failed.